cpp/src/gis/point_in_polygon.cu

```diff
@@ -0,0 +1,184 @@
+ * @param[in] p3_x: Longitude of the third point p3
+ * @param[in] p3_y: Latitude of the third point p3
+ *
+ * @returns positive if it's clockwise, negative if is counter-clockwise and 0 if is colinear
+ */
+template <typename T>
+__device__ T orientation(const T p1_x, const T p1_y, const T p2_x, const T p2_y, const T p3_x, const T p3_y)
+{
+    return ((p2_y - p1_y) * (p3_x - p2_x) - (p2_x - p1_x) * (p3_y - p2_y));
+}
+
+/** 
+ *  @brief Determine whether or not coordinates (query points) are completely inside a static polygon
+ *
+ * @param[in] poly_lats: Pointer to latitudes of a polygon
+ * @param[in] poly_lons: Pointer to longitudes of a polygon
+ * @param[in] point_lats: Pointer to latitudes of many query points
+ * @param[in] point_lons: Pointer to longitudes of many query points
+ * @param[in] poly_size: Size of polygon
+ * @param[in] point_size: Total number of query points
+ * @param[out] point_is_in_polygon: Pointer indicating if the i-th query point is inside or not
+ *
+ * @returns
+ */
+ template <typename T>
+ __global__ void point_in_polygon_kernel(const T* poly_lats,
+                                    const T* poly_lons,
+                                    const T* point_lats,
+                                    const T* point_lons,
+                                    gdf_size_type poly_size,
+                                    gdf_size_type point_size,
+                                    int8_t* point_is_in_polygon)
+{
+    gdf_index_type start_idx = blockIdx.x * blockDim.x + threadIdx.x;
+ 
+    for(gdf_index_type idx = start_idx; idx < point_size; idx += blockDim.x * gridDim.x)
+    {
+        T point_lat = point_lats[start_idx];
+        T point_lon = point_lons[start_idx];
+        gdf_size_type count = 0;
+ 
+        for(gdf_index_type poly_idx = 0; poly_idx < poly_size - 1; poly_idx++) 
+        {
+            if(poly_lons[poly_idx] <= point_lon && point_lon < poly_lons[poly_idx + 1])
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+            else if (point_lon <= poly_lons[poly_idx] && poly_lons[poly_idx + 1] < point_lon) 
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+        }
+        if ((count > 0) && (count % 2 == 0)) point_is_in_polygon[start_idx] = 1;
+        else point_is_in_polygon[start_idx] = 0;
+    }
+}
+
+struct point_in_polygon_functor {
+    template <typename col_type>
+    static constexpr bool is_supported()
+    {
+        return std::is_arithmetic<col_type>::value ;
+    }
+ 
+    template <typename col_type, typename std::enable_if< is_supported<col_type>() >::type* = nullptr>
```

Comment:
Just a little more compact. 

```suggestion
    template <typename col_type, std::enable_if_t< is_supported<col_type>() >* = nullptr>
```

Reply:
Ok, more compact.

Reply:
Also, I added more specifications about the point_in_polygon function in `cpp/include/gis.hpp`